backend/server/routers/courses.py

```diff
@@ -44,77 +62,78 @@ def fixUserData(userData: dict):
-                                "raw_requirements" : "",
-                                "gen_ed": 1,
-                                "path_from": {}
-                            }
-                        }
+                        "study_level": "Undergraduate",
+                        "school": "School of Computer Science and Engineering",
+                        "faculty": "Faculty of Engineering",
+                        "campus": "Sydney",
+                        "equivalents": {"DPST1091": 1, "COMP1917": 1},
+                        "exclusions": {"DPST1091": 1},
+                        "path_to": {
+                            "COMP1521": 1,
+                            "COMP1531": 1,
+                            "COMP2041": 1,
+                            "COMP2111": 1,
+                            "COMP2121": 1,
+                            "COMP2521": 1,
+                            "COMP9334": 1,
+                            "ELEC2117": 1,
+                            "SENG2991": 1,
+                        },
+                        "terms": ["T1", "T2", "T3"],
+                        "raw_requirements": "",
+                        "gen_ed": 1,
+                        "path_from": {},
                     }
                 }
-            })
+            },
+        },
+    },
+)
 def getCourse(courseCode: str):
-    result = coursesCOL.find_one({'code' : courseCode})
+    result = coursesCOL.find_one({"code": courseCode})
     if not result:
-        raise HTTPException(status_code=400, detail=f"Course code {courseCode} was not found")
+        raise HTTPException(
+            status_code=400, detail=f"Course code {courseCode} was not found"
+        )
     result.setdefault("school", None)
-    del result['_id']
+    del result["_id"]
 
     return result
 
+
 @router.get("/searchCourse/{string}")
 def search(string):
-    pat = re.compile(r'{}'.format(string), re.I)
-    code_query = coursesCOL.find({'code': {'$regex': pat}})
-    title_query = coursesCOL.find({'title': {'$regex': pat}})
-
-    return { course['code']: course['title'] for course in chain(code_query, title_query) }
-
-@router.post("/getAllUnlocked/", response_model=CoursesState,
-            responses={
-                400: {"model": message, "description": "Uh oh you broke me"},
-                200: {
-                    "description": "Returns the state of all the courses",
-                    "content": {
-                        "application/json": {
-                            "example": {
-                                "COMP9302": {
-                                    "is_accurate": True,
-                                    "unlocked": True,
-                                    "handbook_note": "This course can only be taken in the final term of your program.",
-                                    "warnings": []
-                                }
-                            }
+    pat = re.compile(r"{}".format(string), re.I)
```

Comment:
Issues with string matching relating to backslashes here. This function should be refactored to string matching.